### PR TITLE
loadSchemaDialects function

### PR DESCRIPTION
**Pull request by HodaSalim**
Made a simple function to expose the _dialects object keys and filtering with hasDialect to make sure that we get the loaded schema dialects for json-schema LSP Project

Co-authored-by: Azzam Uddin <uazzam69@gmail.com> @Azzam1503

### CODE REVIEW

File: lib/keywords.js
Comment (jdesrosiers) [inline lib/keywords.js:85]: The name of this function doesn't make sense. It's not loading dialects.

File: lib/keywords.js
Comment (jdesrosiers) [inline lib/keywords.js:86]: What's the reason for the filter? What is potentially filtered out?

**Review by jdesrosiers — CHANGES_REQUESTED** [review]
Please include tests and documentation (in the README) for your new function.

File: lib/keywords.js
Comment (Azzam1503) [reply lib/keywords.js:85]: what about getDialects?

Comment (jdesrosiers) [reply lib/keywords.js:85]: That's better, but how about "getDialectIds"? It's not the dialect that it's returning, but the identifiers for the dialects.

Comment (Azzam1503) [reply lib/keywords.js:85]: Yeah sounds perfect.

File: README.md
Comment (jdesrosiers) [inline README.md:560]: Please undo this unnecessary change.

File: README.md
Comment (jdesrosiers) [inline README.md:574]: This feels like it's saying the same thing three times. Let's simplify. The last sentence is unnecessary because the types already communicate that it returns an array of strings. We could keep the first or the second sentence, but I don't think we need both. I prefer the first.

```suggestion
    This function retrieves the identifiers of all loaded JSON Schema dialects.
```

File: lib/get-dialect-ids.spec.ts
Comment (jdesrosiers) [inline lib/get-dialect-ids.spec.ts:7]:
```suggestion
test("should return only imported schema in the array if no custom dialects are loaded", () => {
```

File: lib/keywords.js
Comment (jdesrosiers) [inline lib/keywords.js:87]: I prefer the inline style for small one-liners like this.

```suggestion
export const getDialectIds = () => Object.keys(_dialects);
```

File: lib/get-dialect-ids.spec.ts
Comment (jdesrosiers) [inline lib/get-dialect-ids.spec.ts:4]: Let's include all supported dialects to make sure we get only what we expect. (That should include the "stable" and "openapi" dialects.)

File: lib/get-dialect-ids.spec.ts
Comment (jdesrosiers) [inline lib/get-dialect-ids.spec.ts:12]: This test description needs to be better. You describe well what the expectation is, but not under what circumstances you expect that result. Hint: Say something about this testing the custom dialect case.

File: lib/get-dialect-ids.spec.ts
Comment (jdesrosiers) [inline lib/get-dialect-ids.spec.ts:28]: Wrap all these tests in a `describe`. It makes the test output nicer to consume. It should be something that indicates that these tests are related to dialects, vocabularies, and keywords. Keep it as simple as possible.

**Review by jdesrosiers — CHANGES_REQUESTED** [review]

**Comment by jdesrosiers** [comment]
This PR is a mess and can't be reviewed or merged in it's current state. This is why I told you to use `rebase` to update your branch instead of `merge`. You need to drop your merge commit and rebase properly.

File: lib/get-dialect-ids.spec.ts
Comment (jdesrosiers) [inline lib/get-dialect-ids.spec.ts:2]: In tests I try to import only from public APIs. This tests that the function was exported properly.

```suggestion
import { getDialectIds, loadDialect } from "./experimental.js";
```

File: lib/get-dialect-ids.spec.ts
Comment (jdesrosiers) [inline lib/get-dialect-ids.spec.ts:45]: Using `loadDialect` here works, but that's not really a different case than loading dialects using the imports. The case we really want to test is when someone registers a meta-schema (it's considered a meta-schema if it includes the `$vocabulary` keyword).

There doesn't seem to be any value in adding two dialects. I think we're just testing the same thing twice. We can drop one unless there's another distinct case to test.

```suggestion
    registerSchema({
      "$id": "http://example.com/dialect1",
      "$schema": "https://json-schema.org/draft/2020-12/schema",
      "$vocabulary": {
        "https://json-schema.org/draft/2020-12/vocab/core": true,
        "https://json-schema.org/draft/2020-12/vocab/applicator": true
      }
    });
```

File: lib/keywords.js
Comment (jdesrosiers) [inline lib/keywords.js:87]: Please remove this trailing whitespace.

**Review by jdesrosiers — CHANGES_REQUESTED** [review]
Just a couple more details to clean up. We're almost there!

(It looks like some of the commits got squashed in the attempt to clean up after the merge, but it appears that all the content is there.)

File: lib/get-dialect-ids.spec.ts
Comment (Azzam1503) [reply lib/get-dialect-ids.spec.ts]: Done.

File: lib/keywords.js
Comment (Azzam1503) [reply lib/keywords.js]: Done.

**Review by jdesrosiers — APPROVED** [review]
Looks good :tada:. I'll wait until to tomorrow to merge to give @HodaSalim a chance to fix the author meta-data on her commits.

**Pull request merged**
